sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlobTestBase.java

```diff
@@ -0,0 +1,1031 @@
+                return Mono.just(new byte[0]);
+            }
+
+            @Override
+            public Mono<String> getBodyAsString() {
+                return Mono.just("");
+            }
+
+            @Override
+            public Mono<String> getBodyAsString(Charset charset) {
+                return Mono.just("");
+            }
+        };
+    }
+
+    protected void waitForCopy(BlobContainerClient bu, String status) throws Exception {
+        OffsetDateTime start = OffsetDateTime.now();
+        while (!Objects.equals(status, CopyStatusType.SUCCESS.toString())) {
+            status = bu.getPropertiesWithResponse(null, null, null).getHeaders().getValue(X_MS_COPY_STATUS);
+            OffsetDateTime currentTime = OffsetDateTime.now();
+            if (Objects.equals(status, CopyStatusType.FAILED.toString()) || currentTime.minusMinutes(1) == start) {
+                throw new Exception("Copy failed or took too long");
+            }
+            sleepIfRunningAgainstService(1000);
+        }
+    }
+
+    /**
+    * Validates the presence of headers that are present on a large number of responses. These headers are generally
+    * random and can really only be checked as not null.
+    * @param headers The object (may be headers object or response object) that has properties which expose these common headers.
+    * @return Whether or not the header values are appropriate.
+    */
+    protected static boolean validateBasicHeaders(HttpHeaders headers) {
+        return headers.getValue(HttpHeaderName.ETAG) != null
+            // Quotes should be scrubbed from etag header values
+            && !headers.getValue(HttpHeaderName.ETAG).contains("\"")
+            && headers.getValue(LAST_MODIFIED) != null
+            && headers.getValue(X_MS_REQUEST_ID) != null
+            && headers.getValue(X_MS_VERSION) != null
+            && headers.getValue(HttpHeaderName.DATE) != null;
+    }
+
+    protected static boolean validateBlobProperties(Response< BlobProperties > response, String cacheControl,
+        String contentDisposition, String contentEncoding, String contentLanguage, byte[] contentMD5,
+        String contentType) {
+        return Objects.equals(response.getValue().getCacheControl(), cacheControl)
+            && Objects.equals(response.getValue().getContentDisposition(), contentDisposition)
+            && Objects.equals(response.getValue().getContentEncoding(), contentEncoding)
+            && Objects.equals(response.getValue().getContentLanguage(), contentLanguage)
+            && Arrays.equals(response.getValue().getContentMd5(), contentMD5)
+            && Objects.equals(response.getValue().getContentType(), contentType);
+    }
+
+    /**
+     * Insecurely and quickly generates a random AES256 key for the purpose of unit tests. No one should ever make a
+     * real key this way.
+     */
+    protected static byte[] getRandomKey() {
+        long seed = new Random().nextLong();
```

Comment:
This method will need to be made non-static so that they key is the same between operations. Otherwise the assertion of the key SHA256 validation will continue to fail.